### PR TITLE
Implemented versioning from operator-rs.

### PR DESCRIPTION
**Pull request by maltesander**
## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: deploy/crd/nificluster.crd.yaml [hunk 1]
```diff
@@ -147,6 +147,7 @@ spec:
                 version:
                   enum:
                     - 1.13.2
+                    - 1.14.0
```
Comment (razvan) [inline deploy/crd/nificluster.crd.yaml:150]: I thought we don't support version 1.14 yet.

Comment (maltesander) [reply deploy/crd/nificluster.crd.yaml:150]: No but fixing that isnt part of this pull request.

Comment (razvan) [reply deploy/crd/nificluster.crd.yaml:150]: Yes, but then this PR shouldn't add it to the list of supported versions, right?

Comment (maltesander) [reply deploy/crd/nificluster.crd.yaml]: done

**Pull request merged**